### PR TITLE
Support editing cluster name 

### PR DESCRIPTION
**Pull request by liyuheng55555**

### CODE REVIEW

**Review by liyuheng55555 — COMMENTED** [review]
TBD

**Review by liyuheng55555 — COMMENTED** [review]
LGTM

**Review by liyuheng55555 — COMMENTED** [review]
LGTM

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBStartCheck.java [hunk 1]
```diff
@@ -290,8 +285,7 @@ private void throwException(String parameter, Object badValue) throws Configurat
   /** call this method to serialize ClusterName and DataNodeId */
   public void serializeClusterNameAndDataNodeId(String clusterName, int dataNodeId)
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/conf/IoTDBStartCheck.java:286]: .

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties [hunk 2]
```diff
@@ -539,7 +539,7 @@ dn_metric_prometheus_reporter_port=9092
 ####################
 
 # Used for indicate cluster name and distinguish different cluster.
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-system.properties:541]: add comments

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java [hunk 3]
```diff
@@ -86,4 +90,30 @@ public void testSetConfiguration() throws Exception {
...
+    Statement statement = connection.createStatement();
+    // set cluster name on cn and dn
+    statement.execute("set configuration \"cluster_name\"=\"xx\"");
+    ResultSet variables = statement.executeQuery("show variables");
+    variables.next();
+    Assert.assertEquals("xx", variables.getString(2));
+    // restart successfully
+    EnvFactory.getEnv().getDataNodeWrapper(0).stop();
+    EnvFactory.getEnv().getDataNodeWrapper(0).start();
+    Thread.sleep(10000);
```
Comment (OneSizeFitsQuorum) [inline integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java:106]: .

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java [hunk 4]
```diff
@@ -86,4 +90,30 @@ public void testSetConfiguration() throws Exception {
...
+    IConfigNodeRPCService.Iface configNodeClient =
+        EnvFactory.getEnv().getLeaderConfigNodeConnection();
+    TShowClusterResp resp = configNodeClient.showCluster();
+    Assert.assertEquals(NodeStatus.Running.toString(), resp.getNodeStatus().get(1));
+    // set cluster name on datanode, cannot restart
+    statement.executeQuery("set configuration \"cluster_name\"=\"yy\" on 1");
+    EnvFactory.getEnv().getDataNodeWrapper(0).stop();
+    EnvFactory.getEnv().getDataNodeWrapper(0).start();
+    Thread.sleep(10000);
```
Comment (OneSizeFitsQuorum) [inline integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java:115]: .

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java [hunk 5]
```diff
@@ -86,4 +90,30 @@ public void testSetConfiguration() throws Exception {
...
+    resp = configNodeClient.showCluster();
+    Assert.assertEquals(NodeStatus.Unknown.toString(), resp.getNodeStatus().get(1));
```
Comment (OneSizeFitsQuorum) [inline integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java:117]: add configuration file check

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java (hunk 3 above)
Comment (liyuheng55555) [reply integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBSetConfigurationIT.java:106]: Awaitbility

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**